src/cascadia/TerminalApp/JsonUtilsNew.h

```diff
@@ -0,0 +1,434 @@
+            return ::Microsoft::Console::Utils::GuidFromString(til::u8u16(Detail::GetStringView(json)));
+        }
+
+        bool CanConvert(const Json::Value& json)
+        {
+            return json.isString();
+        }
+    };
+
+    // (GUID and winrt::guid are mutually convertible!)
+    template<>
+    struct ConversionTrait<winrt::guid> : public ConversionTrait<GUID>
+    {
+    };
+
+    template<>
+    struct ConversionTrait<til::color>
+    {
+        til::color FromJson(const Json::Value& json)
+        {
+            return ::Microsoft::Console::Utils::ColorFromHexString(Detail::GetStringView(json));
+        }
+
+        bool CanConvert(const Json::Value& json)
+        {
+            return json.isString();
+        }
+    };
+
+    template<typename T, typename TBase>
+    struct EnumMapper
+    {
+        using pair_type = std::pair<std::string_view, T>;
+        T FromJson(const Json::Value& json)
+        {
+            const auto name{ Detail::GetStringView(json) };
+            for (const auto& pair : TBase::mappings)
+            {
+                if (pair.first == name)
+                {
+                    return pair.second;
+                }
+            }
+            // the first mapping is the "Default"
+            return TBase::mappings[0].second;
+        }
+
+        bool CanConvert(const Json::Value& json)
+        {
+            return json.isString();
+        }
+    };
+
+    // FlagMapper is EnumMapper, but it works for bitfields.
+    // It supports a string (single flag) or an array of strings.
+    // Does an O(n*m) search; meant for small search spaces!
+    template<typename T, typename TBase>
+    struct FlagMapper
+    {
+        using pair_type = std::pair<std::string_view, T>;
+        const T* _find(const std::string_view name) const
+        {
+            for (const auto& pair : TBase::mappings)
+            {
+                if (pair.first == name)
+                {
+                    return &pair.second;
+                }
+            }
+            return nullptr;
+        }
+
+        T FromJson(const Json::Value& json)
+        {
+            if (json.isString())
+            {
+                const auto name{ Detail::GetStringView(json) };
+                const auto found{ _find(name) };
+                if (found)
+                {
+                    return *found;
+                }
+            }
+            else if (json.isArray())
+            {
+                T value{};
+                for (const auto& element : json)
+                {
+                    const auto name{ Detail::GetStringView(element) };
+                    const auto found{ _find(name) };
+                    if (found)
+                    {
+                        value |= *found;
+                    }
+                }
+                return value;
+            }
+
+            // If we didn't recognize any flags, the default is _NO FLAGS_
+            return static_cast<T>(0);

```

Comment:
This should be impossible to hit because of `CanConvert` right?

Reply:
If that's the case, FailFast?

Reply:
I'd rather not introduce more failfasts if I can help it. I'm 100% on the side of "don't take down the entire application unless there's nothing you can do to recover"